flink-runtime/src/main/java/org/apache/flink/runtime/jobgraph/jsonplan/JsonPlanGenerator.java

```diff
@@ -170,4 +174,54 @@ public static String generatePlan(
             throw new RuntimeException("Failed to generate plan", e);
         }
     }
+
+    public static String generateStreamGraphJson(
+            StreamGraph sg, Map<Integer, JobVertexID> jobVertexIdMap) {
+        try (final StringWriter writer = new StringWriter(1024)) {
+            try (final JsonGenerator gen = new JsonFactory().createGenerator(writer)) {
+                // start of everything
+                gen.writeStartObject();
+
+                gen.writeArrayFieldStart("nodes");
+
+                // info per vertex
+                for (StreamNode node : sg.getStreamNodes()) {
+                    gen.writeStartObject();
+                    gen.writeStringField("id", String.valueOf(node.getId()));
```

Comment:
can we not just deserialize a node (or a cutdown version) of the node object using Jackson or the like; rather than manually constructing json keys and values?

Reply:
Just like `org.apache.flink.runtime.jobgraph.jsonplan.JsonPlanGenerator#generatePlan`, maybe we should rely on as few third-party frameworks as possible.